src/frontends/pytorch/src/op/flatten.cpp

```diff
@@ -3,59 +3,65 @@
 //
 
 #include "openvino/frontend/pytorch/node_context.hpp"
-#include "openvino/opsets/opset10.hpp"
+#include "openvino/op/add.hpp"
+#include "openvino/op/concat.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/reshape.hpp"
+#include "openvino/op/shape_of.hpp"
+#include "openvino/op/slice.hpp"
+#include "openvino/op/squeeze.hpp"
+#include "openvino/op/unsqueeze.hpp"
 
 namespace ov {
 namespace frontend {
 namespace pytorch {
 namespace op {
 
+using namespace ov::op;
+
 OutputVector translate_flatten(NodeContext& context) {
     auto start_dim = context.const_input<int64_t>(1);
     auto end_dim = context.const_input<int64_t>(2);
 
-    auto shape = std::make_shared<opset10::ShapeOf>(context.get_input(0), element::i32);
-    auto rank_ = std::make_shared<opset10::ShapeOf>(shape, element::i32);
-    auto rank = std::make_shared<opset10::Squeeze>(rank_);
-    // Use opset::If for dim normalization
+    auto shape = std::make_shared<v3::ShapeOf>(context.get_input(0), element::i32);
+    auto rank_ = std::make_shared<v3::ShapeOf>(shape, element::i32);
+    auto rank = std::make_shared<v0::Squeeze>(rank_);
+    // Use opset::If for dim normalization. For now we only have flatten with constant start and end
     auto start_dim_node = context.get_input(1);
     auto end_dim_node = context.get_input(2);
     if (start_dim < 0) {
-        start_dim_node = std::make_shared<opset10::Add>(rank, start_dim_node);
+        start_dim_node = context.mark_node(std::make_shared<v1::Add>(rank, start_dim_node));
     }
```

Comment:
I think we can avoid any restriction to have constant `start_dim`, etc. Can we use Min, Max and Abs operations to avoid any `if`?

Reply:
Not sure how to do it and it is not required for now